### DISCUSSION TITLE
Random number generator for ECP group

### DISCUSSION THREAD

**Discussion by hnsantu554** [strongswan/strongswan#2489] [Q&A]
How to configure random number generator for DH ECP group algorithms?
Is it handled by openssl library? Or any specific configuration need to be enabled?
Please help , thank you

**Comment by tobiasbrunner** [comment]
Not sure what exactly you mean. But the random source does depend on the plugin used (see the docs for the [list of plugins that support the ECP groups](https://docs.strongswan.org/docs/5.9/config/IKEv2CipherSuites.html#_nist_elliptic_curve_groups)). If you use the _openssl_ plugin, then yes, OpenSSL's internal RNG is used.

**Reply by hnsantu554** [reply]
I am using NSIT ecp256 for IKE proposals , which is not from openssl plugin.
We have a requirement where random number should be used for ecp algorithms.
It will be really helpful if can provide me some pointers on enabling random number generation for ecp256 algorithm.

Thanks for the help

**Reply by tobiasbrunner** [reply]
> I am using NSIT ecp256 for IKE proposals , which is not from openssl plugin.

Where is it from then? One of the other plugins?

> We have a requirement where random number should be used for ecp algorithms.

What does that mean?
